app/controllers/campaigns_controller.rb

```diff
@@ -140,7 +140,10 @@ def advertiser_campaign_params
       :name,
       :url,
       creative_ids: [],
-    )
+    ).tap do |whitelisted|
+      new_status = params[:campaign][:status]
+      whitelisted[:status] = new_status if ["paused", "active"].include?(new_status) && ["paused", "active"].include?(@campaign.status)
+    end
```

Comment:
I’m not really a fan of this. Could we abstract this out into an `authorize_` method (see properties controller)? That way we could flash a descriptive error message.

Reply:
100% agree on moving this logic to an authorizer.